process_request/routines.py

```diff
@@ -28,36 +31,40 @@ def get_data(self, item):
                         username=settings.ARCHIVESSPACE["username"],
                         password=settings.ARCHIVESSPACE["password"],
                         repository=settings.ARCHIVESSPACE["repo_id"])
-        obj = aspace.client.get(item)
+        obj = aspace.client.get(item, params={"resolve": ["resource::linked_agents", "ancestors", "top_container", "top_container::container_locations"]})
         if obj.status_code == 200:
-            return obj.json()
+            as_data = {}
+            item_json = obj.json()
+            item_collection = item_json.get("ancestors")[-1].get("_resolved")
+            as_data['creator'] = get_collection_creator(item_json)
+            as_data['restrictions'] = "TK"
+            as_data['restrictions_text'] = "TK"
+            as_data['collection_name'] = item_collection.get("title")
+            if len(item_json.get("ancestors")) > 1:
+                as_data['aggregation'] = item_json.get("ancestors")[0].get("_resolved").get("display_string")
+            else:
+                as_data['aggregation'] = ""
+            as_data['dates'] = get_dates(item_json)
+            as_data['resource_id'] = item_collection.get("id_0")
+            as_data['title'] = item_json.get("display_string")
+            as_data['ref'] = item_json.get("uri")
+            if check_for_instance_type(item_json, "digital_object"):
+                as_data['container'] = ""
+                as_data['barcode'] = ""
+                as_data['location'] = ""
+            else:
+                if check_for_instance_type(item_json, "microform"):
+                    instance = item_json.get("instances")[check_for_instance_type(item_json, "microform")]
+                else:
+                    instance = item_json.get("instances")[0]
+                    top_container_info = instance.get("sub_container").get("top_container").get("_resolved")
```

Comment:
Again, this is stuff that @p-galligan will be working on, but i think this should all get put in another method called "get_preferred" or something like that.